minoc/src/main/java/org/opencadc/minoc/GetAction.java

```diff
@@ -247,4 +282,51 @@ boolean hasNoOperations() {
             return conflicts;
         }
     }
+
+    /**
+     * Range is outside the file content
+     */
```

Comment:
Partial review only because this part will reshape the other code.

This exception must be defined in cadc-util (as RangeNotSatisfiableException to match the status code definition) *and* added to HttpTransfer.checkErrors to map a 416 to this exception. That can be reviewed and published.

Once that is done, it must be added to cadc-rest RestAction's main catch that maps exceptions to codes (416 here). Then this code simply throws that exception.

Also, I think you need to set some of the headers from `HeaderAction.setHeaders`. In my put-txn work I am adding a `HeaderAction.setTransactionHeaders` to supplement the standard headers... here it might be more subtle than that so you might have to figure out a clean way to not set all the standard headers if some should not be set, but it looks like all are applicable/unchanged.

Reply:
Didn't know about this design rule. A bit weird that it's not only `HttpGet` that uses that exception.
PR: https://github.com/opencadc/core/pull/169